src/v2/manifest/mod.rs

```diff
@@ -19,62 +19,68 @@ impl Client {
     /// The name and reference parameters identify the image.
     /// The reference may be either a tag or digest.
     pub fn get_manifest(&self, name: &str, reference: &str) -> FutureManifest {
-        let url = match hyper::Uri::from_str(&format!(
-            "{}/v2/{}/manifests/{}",
-            self.base_url.clone(),
-            name,
-            reference
-        )) {
-            Ok(url) => url,
-            Err(e) => {
-                let msg = format!("failed to parse Uri from str: {}", e);
-                error!("{}", msg);
-                return Box::new(futures::future::err::<_, _>(Error::from(msg)));
-            }
-        };
-        let req = {
-            let mut req = match self.new_request(hyper::Method::GET, url.clone()) {
-                Ok(r) => r,
+        Box::new(
+            self.get_manifest_and_ref(name, reference)
+                .map(|(manifest, _)| manifest),
+        )
+    }
+
+    /// Fetch an image manifest and return it with its digest.
+    ///
+    /// The name and reference parameters identify the image.
+    /// The reference may be either a tag or digest.
+    pub fn get_manifest_and_ref(&self, name: &str, reference: &str) -> FutureManifestAndRef {
+        let url = {
+            let ep = format!(
+                "{}/v2/{}/manifests/{}",
+                self.base_url.clone(),
+                name,
+                reference
+            );
+            match reqwest::Url::parse(&ep) {
+                Ok(url) => url,
                 Err(e) => {
-                    let msg = format!("new_request failed: {}", e);
-                    error!("{}", msg);
-                    return Box::new(futures::future::err(Error::from(msg)));
+                    return Box::new(futures::future::err::<_, _>(Error::from(format!(
+                        "failed to parse url from string '{}': {}",
+                        ep, e
+                    ))));
                 }
-            };
-            let mtype = mediatypes::MediaTypes::ManifestV2S2.to_string();
-            req.headers_mut().append(
-                header::ACCEPT,
```

Comment:
Did you drop this logic on purpose?

Reply:
No, though it doesn't seem required. Should I add it back?

Reply:
Still unsure but I added it back to keep the semantics unchanged

Reply:
This is a "SHOULD" according to registry spec. I don't remember exactly, but I think docker hub may require this.